h/static/scripts/annotator/anchoring/pdf.coffee

```diff
@@ -0,0 +1,224 @@
+seek = require('dom-seek')
+
+Annotator = require('annotator')
+xpathRange = Annotator.Range
+
+html = require('./html')
+{TextPositionAnchor, TextQuoteAnchor} = require('./types')
+
+
+getSiblingIndex = (node) ->
+  siblings = Array.prototype.slice.call(node.parentNode.childNodes)
+  return siblings.indexOf(node)
+
+
+getNodeTextLayer = (node) ->
+  until node.classList?.contains('page')
+    node = node.parentNode
+  return node.getElementsByClassName('textLayer')[0]
+
+
+getPage = (pageIndex) ->
+  return PDFViewerApplication.pdfViewer.pages[pageIndex]
+
+
+getPageTextContent = (pageIndex, cache) ->
+  if cache[pageIndex]?
+    return Promise.resolve(cache[pageIndex])
+  else
+    joinItems = ({items}) ->
+      nonEmpty = (item.str for item in items when /\S/.test(item.str))
```

Comment:
If you know more about the rationale behind doing this, that would be another candidate for a comment, IMO. It's not immediately clear to me when reading this code why we're doing this normalisation/sanitisation step, and it's not clear that it wouldn't potentially break if we got back something like

    <TextItem str="hello"><TextItem str=" "><TextItem str="world">

which would presumably be completely valid from PDF.js's point of view? (I don't know, which is why I'm asking...)